### PR TITLE
chore: add test to http request without specifying port

### PR DESCRIPTION
**Pull request by morenol**

### CODE REVIEW

File: crates/wasi-http/tests/all/main.rs [hunk 1]
```diff
@@ -527,6 +527,29 @@ async fn do_wasi_http_echo(uri: &str, url_header: Option<&str>) -> Result<()> {
...
 
+#[test_log::test(tokio::test)]
+async fn wasi_http_without_port() -> Result<()> {
+    do_http_request_without_port().await
+}
+
+async fn do_http_request_without_port() -> Result<()> {
+    let request = hyper::Request::builder()
+        .method(http::Method::GET)
+        .uri("http://example.com/test-path");
```
Comment (elliottt) [inline crates/wasi-http/tests/all/main.rs:538]: Could you add an additional request for a path that will return a 200?

Comment (morenol) [reply crates/wasi-http/tests/all/main.rs:538]: Sure, for some reason this is returning `404` in the CI, any idea why?

**Comment by iawia002** [comment]
> Thanks! To land this though it'll need to pass CI, so mind folding this into #8671?

I am back online, @morenol do you mind if I cherry-pick this commit into #8671 and square them into one commit? I hope we can land this fix soon.

**Comment by morenol** [comment]
I don't mind but I think that this test is not accurate, feel free to check and create a test that could reproduce the current issue

**Comment by morenol** [comment]
Closed since a similar test was added in #8671

**Pull request closed without merging**